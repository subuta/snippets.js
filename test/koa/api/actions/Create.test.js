/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import CreateAction from 'lib/koa/api/actions/Create'

describe('koa/api/actions/Create', () => {
  it('should create koa action', () => {
    expect(format(CreateAction({
      model: 'user',
      routeConfig: {}
    }))).toMatchSnapshot()
  })
})
