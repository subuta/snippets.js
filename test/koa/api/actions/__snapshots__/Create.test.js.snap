// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`koa/api/actions/Create should create koa action 1`] = `
"user.post('/', async (ctx) => {
  const {User} = ctx.state.models
  const {user} = ctx.request.body

  let params = {}

  /* mat Before create [start] */
  /* mat Before create [end] */

  let response = await User.query()
    .insert({
      ...user,
      ...params
    })
    .eager('')

  /* mat After create [start] */
  /* mat After create [end] */

  ctx.body = response
})
"
`;
